### ISSUE TITLE
Provide a clearer message if a user connects to a DartVM that isn't connected to Flutter Tools.

### ISSUE THREAD

**Issue by jacob314** [flutter/devtools#4091]
From a user on discord: 

> Today, i tried something new...
> i ran an android app with a view in flutter (add-to-app) and observed that when the FlutterEngine is initialised, i get a log saying flutter: Observatory listening on http://127.0.0.1:38177/<some alphanumeric>/. I tried to open this in Flutter Devtools (by installing Termux, exposing this port to the internet using ngrok and opening this in Devtools) but got a message saying "This application does not have any tabs".
> What is different when the app is run using flutter run and when the debug apk is run? if there is no way to open the observatory, why the log?

The problem is the Dart VM likely cannot evaluate expressions as Flutter tools isn't attached to it. We see this combined with a VM that appears to be debug mode and get confused and show zero tabs instead of showing a message like "VM Service not fully initialized, run `flutter attach` so that you can debug.

**Comment by jacob314** [comment]
We've also seen some g3 users hit similar edge cases where the VMService crashed when in this state so having a clearer message in DevTools would be nice.

**Comment by kenzieschmoll** [comment]
Is the use case (and the bug) described here still valid today? @bkonyi

**Comment by bkonyi** [comment]
Yeah, I think so. If you're connecting directly to the VM service URI without using `flutter attach` (or possibly even _with_ `flutter attach`, I'm not 100% sure off the top of my head), there won't be a resident frontend server available to perform compilations for expression evaluation.
